test/integration/controller/jobs/job/job_controller_test.go

```diff
@@ -2109,6 +2109,8 @@ var _ = ginkgo.Describe("Job controller interacting with Workload controller whe
 						Message: "The workload is deactivated due to exceeding the maximum number of re-queuing retries",
 					}, util.IgnoreConditionTimestampsAndObservedGeneration),
 				))
+				// Fake client that are using on unit tests doesn't allow us to check it.
+				g.Expect(apimeta.FindStatusCondition(wl.Status.Conditions, kueue.WorkloadDeactivationTarget)).Should(gomega.BeNil())
```

Comment:
Use `g.Expect(wl.Status.Conditions).NotTo(gomega.ContainElements`

Reply:
In this case it is enough if the message is changed and the test will pass even if the condition stays.

Reply:
you can add `cmpopts.IgnoreFields`

Reply:
you know, maybe this is fine. It's more problematic if you use booleans. But since FindStatusCondition returns the full condition, it's obvious what the problem is if this check fails.